action/deregister.go

```diff
@@ -0,0 +1,72 @@
+package action
+
+import (
+	"errors"
+	"io/ioutil"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	"github.com/allegro/vaas-registration-hook/k8s"
+	"github.com/allegro/vaas-registration-hook/vaas"
+)
+
+const (
+	// DeregisterName is the CLI name of this action
+	DeregisterName = "deregister"
+	// FlagBackendID represents a known backend id that is to be deregistered
+	FlagBackendID = "backend-id, id"
+)
+
+// DeregisterCLI removes a backend from VaaS using CLI data
+func DeregisterCLI(c *cli.Context) error {
```

Comment:
Why do we even bother with, cli flags, for every value that should be put into annotations, inside podInfo. Could we give up on cli based code and make this hook, to be based only on annotations/labels?

Reply:
CLI mode makes the hook more universal and not just K8s specific. Plus it makes it possible to test VaaS routines outside K8s.